backend/api/views/factories_cr.py

```diff
@@ -161,6 +183,9 @@ def _handle_create_ohshown_events(request):
             factory=new_factory, report_record=report_record
         )
         Reporter.objects.create(**new_reporter_field)
+        if post_body.get("type") == '2-1':
```

Comment:
Hmmm this part is interesting. We may think about how to make it more readable.

Reply:
Use something like enum?
Also type '2-1' or '2-2' kind of redundant.
Integer + enum feels good to me.